src/components/username/Username.tsx

```diff
@@ -0,0 +1,114 @@
+import { useAppDispatch } from "@hooks";
+import { getScrobbles, getUserInfo } from "api/lastfm";
+import classNames from "classnames/bind";
+import { addScrobbles } from "components/upload/uploadSlice";
+import { Track } from "models/ScrobblePage";
+import { User } from "models/User";
+import { FormEvent, useRef, useState } from "react";
+import styles from './Username.module.scss';
+import { setUsername as setUsernameRedux } from "./userSlice";
+
+const cx = classNames.bind(styles);
+
+function UsernameInput() {
+  const [username, setUsername] = useState('');
+  const [finished, setFinished] = useState(false);
+
+  const setTotalCSV = (csv: Track[]) => {
+    console.log('all', csv)
+  }
+
+  const dispatch = useAppDispatch();
+
+  const handleFormSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    getUserInfo(username).then((info: User) => {
+      dispatch(setUsernameRedux(info))
+      getUserLastTracks(username, info)
+    })
+  }
+
+  const getUserLastTracks = (
+    async (user: string, info: User) => {
+      if (finished) {
```

Comment:
For some reason, this isn't updated when changing between usernames. Check logic flow.